src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -558,3 +558,32 @@ case object RETURN extends OpCode(0xf3, 2, 0, G_zero) {
     calcMemCost(state.memory.size, addr, size)
   }
 }
+
+case object SELFDESTRUCT extends OpCode(0xff, 1, 0, G_selfdestruct) {
+  protected def exec(state: ProgramState): ProgramState = {
+    val (refundDW, stack1) = state.stack.pop
+    val refundAddr: Address = Address(refundDW)
+    val ownerAccount: Account = state.context.account
+    val ownerAddr: Address = state.context.env.ownerAddr
+    val gasRefund: BigInt = if (state.addressesToDelete contains ownerAddr) 0 else R_selfdestruct
```

Comment:
It only deals with the refund. There's no mechanism to prevent a subsequent message call to this account from executing its current code.